tests/benchmarks/micro_benchmarks/test_flops_performance.py

```diff
@@ -0,0 +1,88 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""Tests for flops-perf benchmark."""
+
+import os
+import stat
+from pathlib import Path
+
+from superbench.benchmarks import BenchmarkRegistry, Platform, BenchmarkType
+
+
+def test_flops_performance_cuda():
+    """Test flops-perf benchmark."""
+    benchmark_name = 'flops-perf'
+    (benchmark_class,
+     predefine_params) = BenchmarkRegistry._BenchmarkRegistry__select_benchmark(benchmark_name, Platform.CUDA)
+    assert (benchmark_class)
+
+    benchmark = benchmark_class(
+        benchmark_name, parameters='--num_warmup 200 --n 1024 --k 512 --m 2048 --precision FP32 TF32_TC FP16_TC'
+    )
+
+    # Create fake binary file just for testing.
+    binary_path = os.path.join(os.getenv('SB_MICRO_PATH', '/usr/local'), 'bin')
+    binary_path = os.path.join(binary_path, 'cutlass_profiler')
+    Path(binary_path).touch(exist_ok=True)
+    st = os.stat(binary_path)
+    os.chmod(binary_path, st.st_mode | stat.S_IEXEC)
+
+    assert (benchmark._preprocess())
+
+    # Check basic information.
+    assert (benchmark.name == 'flops-perf')
+    assert (benchmark.type == BenchmarkType.MICRO)
+    assert (benchmark._bin_name == 'cutlass_profiler')
+
+    # Check parameters specified in BenchmarkContext.
+    assert (benchmark._args.num_warmup == 200)
+    assert (benchmark._args.n == 1024)
+    assert (benchmark._args.k == 512)
+    assert (benchmark._args.m == 2048)
+    assert (benchmark._args.precision == ['FP32', 'TF32_TC', 'FP16_TC'])
+
+    # Check the command list.
+    for i in range(len(benchmark._args.precision)):
+        command = '{} --warmup-iterations={} --operation=gemm --n={} --k={} --m={} --kernels={}'.format(
+            benchmark._bin_name, benchmark._args.num_warmup, benchmark._args.n, benchmark._args.k, benchmark._args.m,
+            benchmark._FlopsPerformanceCuda__kernel_map[benchmark._args.precision[i]]
+        )
+        expected_cmd = benchmark._bin_name + benchmark._commands[i].split(benchmark._bin_name)[1]
+        assert (command == expected_cmd)
+
+    # Check results and metrics.
+    raw_output_FP32 = """
```

Comment:
is there any failure case to test?

Reply:
For output parsing, if GLOPs can't be extracted from the raw text, the value will be set as 0. I can add one test case for this scenario.

Reply:
Have added more cases.